src/main/java/seedu/todo/ui/InputHandler.java

```diff
@@ -1,12 +1,95 @@
 package seedu.todo.ui;
 
+import java.util.LinkedList;
+
 import seedu.todo.controllers.*;
 
 public class InputHandler {
+    
+    private static InputHandler instance;
+    
+    private static final int MAX_HISTORY_SIZE = 20;
+    private static final int INITIAL_HISTORY_INDEX = -1;
+    private static LinkedList<String> commandHistory = new LinkedList<String>();
```

Comment:
As in I'm saying the left side should be `Deque<String> commandHistory = ...` and the right side can be LinkedList, but I'm not really sure which is faster =x

Maybe just keep it as LinkedList for now, it's a trivial one-line fix if we want to change